### DISCUSSION TITLE
Easy way to iterate over trades?

### DISCUSSION THREAD

**Discussion by flexelem** [kernc/backtesting.py#556] [Q&A]
Hi there,

I want to persist each trade my strategy is applying. I am getting trades from stats object returned from backtest.run() method. I couldn't figure out an easy way to iterate over the list trades instead I found myself trying to figure out getting values of each attribute like the way below;

```
bt = Backtest(df,
              WTCross,
              cash=10000,
              commission=.002,
              exclusive_orders=True,
              trade_on_close=True)
stats = bt.run()

print(stats)
print(stats._strategy)
print(stats._trades)

for ind in range(len(stats._trades)):
    trade_row = stats._trades.iloc[[ind]]
    print(trade_row['ExitPrice'])
```

My question is, is there a better way to get the list of trades?

Thanks,
Burak

**Comment by kernc** [comment]
`stats._trades` is a pandas DataFrame, so you're looking for how to [iterate over DataFrame rows](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iterrows.html) ...

**Reply by flexelem** [reply]
Hi @kernc 

Sorry for my late response. I will try thanks!
